File: Games/Sevens/src/main/kotlin/jokatu/game/games/sevens/views/SevensViewResolverFactory.java
package jokatu.game.games.sevens.views;

import jokatu.game.games.sevens.game.SevensGame;
import jokatu.game.games.sevens.player.SevensPlayer;
import jokatu.game.viewresolver.ViewResolver;
import jokatu.game.viewresolver.ViewResolverFactory;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

/**
 * @author steven
 */
@Component
public class SevensViewResolverFactory extends ViewResolverFactory<SevensPlayer, SevensGame> {
	@NotNull
	@Override
	protected Class<SevensGame> getGameClass() {
		return SevensGame.class;
	}

	@NotNull
	@Override
	protected ViewResolver<SevensPlayer, SevensGame> getResolverFor(@NotNull SevensGame castGame) {
		return new SevensViewResolver(castGame);
	}
}

File: Games/Sevens/src/main/kotlin/jokatu/game/games/sevens/views/SevensViewResolverFactory.kt
package jokatu.game.games.sevens.views

import jokatu.game.games.sevens.game.SevensGame
import jokatu.game.games.sevens.player.SevensPlayer
import jokatu.game.viewresolver.ViewResolver
import jokatu.game.viewresolver.ViewResolverFactory
import org.springframework.stereotype.Component

/**
 * @author steven
 */
@Component
class SevensViewResolverFactory : ViewResolverFactory<SevensPlayer, SevensGame>() {
	override val gameClass: Class<SevensGame>
		get() = SevensGame::class.java

	override fun getResolverFor(castGame: SevensGame): ViewResolver<SevensPlayer, SevensGame> {
		return SevensViewResolver(castGame)
	}
}
